plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -561,7 +627,14 @@ protected JdbcOutputTableHandle createTable(ConnectorSession session, ConnectorT
             List<ColumnMetadata> columns = tableMetadata.getColumns();
             ImmutableList.Builder<String> columnNames = ImmutableList.builderWithExpectedSize(columns.size());
             ImmutableList.Builder<Type> columnTypes = ImmutableList.builderWithExpectedSize(columns.size());
-            ImmutableList.Builder<String> columnList = ImmutableList.builderWithExpectedSize(columns.size());
+            // columnList is only used for createTableSql - the extraColumns are not included on the JdbcOutputTableHandle
+            ImmutableList.Builder<String> columnList = ImmutableList.builderWithExpectedSize(columns.size() + extraColumns.size());
+
+            for (ColumnMetadata column : extraColumns) {
+                String columnName = identifierMapping.toRemoteColumnName(connection, column.getName());
```

Comment:
It look you always go through `identifierMapping` when accessing `handle.getPageSinkIdColumnName`. Can't you store the mapped name in the `JdbcOutputTableHandle` already?

Then you could drop some args here and there. E.g. `Session` and `Connection` from `buildInsertSql`. Actually, session is not used even now.